### PR TITLE
Fix, upgrade to jquery 3

### PR DESCRIPTION
**Pull request by aboganas**
This PR upgrades .js files to jquery 3; it also upgrade select2 to the latest v 4.0.12 which is compatible with jquery 3

### CODE REVIEW

File: flask_appbuilder/static/appbuilder/js/ab.js [hunk 1]
```diff
@@ -45,25 +45,25 @@ function loadSelectData() {
...
     $(".my_select2").select2({placeholder: "Select a State", allowClear: true});
     loadSelectData();
     loadSelectDataSlave();
-    $(".my_select2.readonly").select2("readonly",true);
+    $(".my_select2.readonly").attr("readonly", "readonly");
     $("a").tooltip({container:'.row', 'placement': 'bottom'});
 });
 
 
 $( ".my_change" ).on("change", function(e) {
```
Comment (aboganas) [inline flask_appbuilder/static/appbuilder/js/ab.js]: still not sure to what this function is attached?

File: flask_appbuilder/static/appbuilder/js/ab.js [hunk 2]
```diff
@@ -45,25 +45,25 @@ function loadSelectData() {
...
- var theForm=document.getElementById("model_form");
-  theForm.action = "";
-  theForm.method = "get";
-  theForm.submit();
- })
+    var theForm=document.getElementById("model_form");
```
Comment (aboganas) [inline flask_appbuilder/static/appbuilder/js/ab.js]: still not sure where this function is attached?

**Comment by aboganas** [comment]
@dpgaspar did you have a chance to review it?

**Comment by dpgaspar** [comment]
Not yet, just finishing 2.2.1 with fixes, but I certainly want to include this on the next release, probably 2.3.0

**Comment by aboganas** [comment]
Sorry for the delay. I will look into that and update the PR accordingly

**Comment by aboganas** [comment]
@dpgaspar I've applied bootstrap theme. I think this PR is now ready.

**Comment by dpgaspar** [comment]
Would love to merge this, but just found a small final problem, on dropdowns with multiple values, the already existing values get repeated has available options

image

@aboganas @nytai

**Comment by nytai** [comment]
@aboganas Looks like this can be fixed by adding the following css to ab.css
```css
.select2-results__option[aria-selected=true] {
    display: none;
}
```

from: https://stackoverflow.com/a/36687069

**Comment by dpgaspar** [comment]
@aboganas would be awesome if you could solve the conflicts so I can merge the PR. Thank you!

**Comment by dpgaspar** [comment]
Included on #1300

**Pull request closed without merging**